substrate/primitives/keyring/src/ed25519.rs

```diff
@@ -45,6 +44,8 @@ pub enum Keyring {
 	Two,
 }
 
+use hex_literal::hex;
```

Comment:
> `hex_literal` crate crate has been removed by substrate in favor of using `array_bytes`.
> 
> But I also understand that here you require an `&'static [u8; 32]`.

Yes, exactly this is a reason.

> Here the `public_bytes()` and `public()` are a bit redundant. Maybe we can do this without having to introduce the `public_bytes()` function but:
> 
>     * gate behind `full_crypto` the `pair`, `sign` and `public` (the one which uses `pair()`) functions

So I think the whole point of this `lazy_static(hashmaps)` was to avoid calling `pair()` which is probably at some cost. I don't know the reason/origin of this optimization, but I am in favor of keeping it. Also because we need to provide alternative implementation for deployments with no `full_crypto` available. Keeping single code for keyring public keys for all feature sets makes sense for me.

> In this case you'll need to remove stuff like:
> 
> ```
> impl From<Keyring> for &'static [u8; 32] {
> ```
> 
> Which I'm not really sure is super relevant

Yeah, we can remove `hex_literal`, but we will need to remove some conversions to refs. Maybe they are not needed at all. Actually I had a version with `array_bytes::hex2bytes_unchecked`.

I'll re-iterate.